src/annotator/plugin/test/pdf-test.js

```diff
@@ -61,17 +69,38 @@ describe('annotator/plugin/pdf', () => {
     $imports.$restore();
   });
 
+  function pdfViewerHasClass(className) {
+    return fakePDFViewerApplication.pdfViewer.viewer.classList.contains(
+      className
+    );
+  }
+
   describe('#constructor', () => {
     it('adds CSS classes to override PDF.js styles', () => {
       pdfPlugin = createPDFPlugin();
-      assert.isTrue(
-        fakePDFViewerApplication.pdfViewer.viewer.classList.contains(
-          'has-transparent-text-layer'
-        )
-      );
+      assert.isTrue(pdfViewerHasClass('has-transparent-text-layer'));
     });
   });
 
+  it('hides annotation layers when there is a text selection', async () => {
+    // This tests checks for a CSS class on the root PDF viewer element.
+    // The annotation layers are hidden by a CSS rule that uses this class.
+    const selection = window.getSelection();
+    if (!selection.isCollapsed) {
+      selection.collapseToStart();
+    }
+    pdfPlugin = createPDFPlugin();
+    assert.isFalse(pdfViewerHasClass('is-selecting'));
+
+    selection.selectAllChildren(document.body);
+    await awaitEvent(document, 'selectionchange');
+    assert.isTrue(pdfViewerHasClass('is-selecting'));
+
+    selection.collapseToStart();
+    await awaitEvent(document, 'selectionchange');
+    assert.isFalse(pdfViewerHasClass('is-selecting'));
+  });
```

Comment:
`awaitEvent` is needed here because the `selectionchange` event is fired asynchronously by the browser. We _could_ forcibly dispatch it ourselves in the test, but it's more faithful to normal application behavior this way.